### PR TITLE
sql: add json{,b}_extract_path_text

### LINKED ISSUE #60569: sql: support json_extract_path_text and jsonb_extract_path_text
The Golang ORM GORM generates queries that use this function when dealing with a JSON column. We already have `json_extract_path` and the `#>>` operator, so we should be able to add this too. See https://www.postgresql.org/docs/current/functions-json.html

---

## Instructions

* Refer to the commit that added `json_extract_path`: https://github.com/cockroachdb/cockroach/pull/20520/commits/66ca0ae0185ec06e45e3bac98dd52c010d37787a. The change here will be fairly similar.
* Modify `pkg/sql/sem/builtins/builtins.go` and add `json_extract_path_text` and `jsonb_extract_path_text`. Reuse the existing code for `json_extract_path`. The only difference should be the return type.
* Add tests in `pkg/sql/logictest/testdata/logic_test/pg_catalog ` and `pkg/sql/logictest/testdata/logic_test/json_builtins`. The tests should be identical to all the `json_extract_path` tests.

### PR DESCRIPTION
**Pull request by kakshay21**
Fixes: #60569
Release note (sql change): Added json_extract_path_text and jsonb_extract_path_text builtins.

### CODE REVIEW

File: pkg/sql/sem/builtins/builtins.go [hunk 1]
```diff
@@ -5579,6 +5583,38 @@ var jsonExtractPathImpl = tree.Overload{
...
+			}
+			if v == tree.DNull {
+				return tree.DNull, nil
+			}
+			path[i-1] = string(tree.MustBeDString(v))
+		}
+		result, err := json.FetchPath(j.JSON, path)
+		if err != nil {
+			return nil, err
+		}
```
Comment (rafiss) [inline pkg/sql/sem/builtins/builtins.go:5604]: can you extract lines 5591-5604 into a function that is shared between jsonExtractPathTextImpl and jsonExtractPathImpl?

**Review by rafiss — CHANGES_REQUESTED** [review]
thank you for this change! i have a small request for a refactor

File: pkg/sql/sem/builtins/builtins.go (hunk 1 above)
Comment (kakshay21) [reply pkg/sql/sem/builtins/builtins.go:5583]: right! missed it 😅

**Review by rafiss — APPROVED** [review]
this looks good. thank you!

bors r+

**Pull request merged**